Source/Core/AudioCommon/WaveFile.cpp

```diff
@@ -38,6 +38,15 @@ bool WaveFileWriter::Start(const std::string& filename, unsigned int HLESampleRa
 
 	audio_size = 0;
 
+	if (basename.empty())
+	{
+		std::string filepath = File::GetUserPath(D_DUMPAUDIO_IDX);
+		basename = filename;
+		basename.replace(0, filepath.length(), "");
```

Comment:
You're initializing `basename` with `filename`, then remove `0 .. filename.length()` from `basename`? That leaves you with an empty string, unless I missed something...